src/hooks/useCheckURL.ts

```diff
@@ -41,7 +44,88 @@ function useCheckURL(urlToCheck: string): {
 			throw new Error("User handle could not be extracted from keystore");
 		}
 		const u = new URL(urlToCheck);
-		if (u.protocol === 'openid-credential-offer' || u.searchParams.get('credential_offer') || u.searchParams.get('credential_offer_uri') ) {
+		if (u.searchParams.get('credential_offer_uri') ) {
+			const credentailOfferResponse = await fetch(u.searchParams.get('credential_offer_uri'));
+			const credentialOffer = await credentailOfferResponse.json();
+
+			if (!credentialOffer.grants['urn:ietf:params:oauth:grant-type:pre-authorized_code'] || !credentialOffer.credential_issuer) {
+				return;
+			}
+
+			const body = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:pre-authorized_code',
+				'pre-authorized_code': credentialOffer.grants['urn:ietf:params:oauth:grant-type:pre-authorized_code']['pre-authorized_code'],
+			};
+
+			const tokenResponse = await fetch(`${credentialOffer.credential_issuer}/token`, {
+				method: 'POST',
+				body: JSON.stringify(body),
+				headers: {
+					'Content-Type': 'application/json',
+				}
+			});
+			const { access_token: accesToken, c_nonce: cNonce } = await tokenResponse.json();
+
+			const openIdCredentialIssuerResponse = await fetch(`${credentialOffer.credential_issuer}/.well-known/openid-credential-issuer`, {
+				headers: {
+					'Cache-Control': 'no-cache',
+				},
+			});
+			const openIdCredentialIssuer = await openIdCredentialIssuerResponse.json();
+			// const metadata = OpenidCredentialIssuerMetadataSchema.parse(openIdCredentialIssuer);
+			const metadata = openIdCredentialIssuer; // @todo: investigate schema errors
+
+			const generateNonceProof = async (cNonce: string, audience: string, clientId: string): Promise<{ jws: string }> => {
```

Comment:
This is copy-pasted from an injected dependency. It would be better to inject the same dependency into some sort of `PreAuthorizedCodeFlowClient`.